### PR TITLE
[fix]fix buf alloc，the size shoud be MAX_N_CMPS * aligned_dim * sizeof(T)

### PR DESCRIPTION
**Pull request by plutolove**
[fix]fix buf malloc，the size shoud be MAX_N_CMPS * aligned_dim * sizeof(T)

### CODE REVIEW

File: src/pq_flash_index.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ namespace diskann {
 
   template<typename T>
   QueryScratch<T>::QueryScratch(size_t aligned_dim, size_t visited_reserve) {
-    _u64 coord_alloc_size = ROUND_UP(MAX_N_CMPS * aligned_dim, 256);
+    _u64 coord_alloc_size = ROUND_UP(MAX_N_CMPS * aligned_dim * sizeof(T), 256);
```
Comment (gopalrs) [inline src/pq_flash_index.cpp:68]: In the DLVS code, we removed the MAX_N_CMPS from the expression to save space. IIRC, coord_scratch only holds one point at a time. Maybe you want to do the same here? 
In the DLVS code, it is simply aligned_dim * sizeof(T).

**Review by gopalrs — CHANGES_REQUESTED** [review]

File: src/pq_flash_index.cpp (hunk 1 above)
Comment (plutolove) [reply src/pq_flash_index.cpp:68]: That is a Bug, If buf_idx > 4069 will coredump

Comment (harsha-simhadri) [reply src/pq_flash_index.cpp:68]: @gopalrs Let's take the reduction of the MAX_N_CMPS as a separate issue.

**Review by harsha-simhadri — APPROVED** [review]

**Comment by harsha-simhadri** [comment]
Replaced with PR #220

**Pull request closed without merging**